soccer/radio/NetworkRadio.cpp

```diff
@@ -0,0 +1,187 @@
+#include "NetworkRadio.hpp"
+#include <Utils.hpp>
+#include "Geometry2d/Util.hpp"
+#include "status.h"
+
+using namespace boost::asio;
+using ip::udp;
+
+NetworkRadio::NetworkRadio(int server_port, int robot_port)
+    : _socket(_context, udp::endpoint(udp::v4(), server_port)),
+      _server_port{server_port},
+      _robot_port{robot_port} {}
+
+NetworkRadio::~NetworkRadio() {}
+
+bool NetworkRadio::open() {
+    startReceive();
+    return true;
+}
+
+void NetworkRadio::startReceive() {
+    // Set a receive callback
+    _socket.async_receive_from(
+        boost::asio::buffer(_recv_buffer), _robot_endpoint,
+        [this] (const boost::system::error_code& error, std::size_t num_bytes) {
+            receivePacket(error, num_bytes);
+        });
+}
+
+bool NetworkRadio::isOpen() const {
+    return _socket.is_open();
+}
+
+void NetworkRadio::send(Packet::RadioTx& packet) {
+    // Get a list of all the IP addresses this packet needs to be sent to
+    for (int robot_idx = 0; robot_idx < packet.robots_size(); robot_idx++) {
+        const Packet::Control& control = packet.robots(robot_idx).control();
+        uint32_t robot_id = packet.robots(robot_idx).uid();
+
+        // Build the control packet for this robot.
+        std::array<uint8_t, rtp::HeaderSize + sizeof(rtp::RobotTxMessage)> forward_packet;
```

Comment:
Since we will have to change everything in rtp.hpp, we should just use the `ForwardSize` defined there instead of `HeaderSize + sizeof(...)`.

This does remind me that we have to change all of that when we change radios

Reply:
Uh, why do we have to change any of it? It's just a struct packing, none of it looks platform specific.

Reply:
My b, I mean to imply that we would have to glance through and change a few things here and there. Most of it is good, but we are changing what information we are sending to the robots. Specifically, we are sending information on a per robot basis instead of all the robots' information to all the robots. I think it'll only be [this line](https://github.com/RoboJackets/robocup-fshare/blob/3dba0305a600eccadfdf80c4a22995e384860654/include/rc-fshare/rtp.hpp#L116), but I haven't gone through and really checked to be sure.

Reply:
Gotcha. Yeah, that's why I wasn't using `ForwardSize` anywhere.